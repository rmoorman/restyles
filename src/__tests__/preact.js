/* eslint-env jest */
/* eslint-disable jsx-quotes */
// eslint-disable-next-line no-unused-vars
import {h} from 'preact' /* @jsx h */
import renderToString from 'preact-render-to-string'
import Style from '../index'
import { matcher, serializer } from 'jest-glamor-react'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)
// options.functions = false
// chai.use(assertJsx)

describe('restyles', () => {
  test('renders preact elements with styles', () => {
    const props = {
      online: true,
      theme: {radius: 5}
    }

    let _class = ''
    const tree = (
      <Style
        css={{
          color: 'preact-purple',
          ':hover': {
            color: 'red'
          },
          '.preact': {
            color: props.online ? 'green' : 'gray',
            fontSize: 20
          }
        }}
      >
        {cls => {
          _class = cls
          return (
            <div className={cls}>
              This will be preact-purple until hovered.
              <div className="profile">
                This font size will be 20px
              </div>
            </div>
          )
        }}
      </Style>
    )

    expect(renderToString(tree)).toContain(_class)
    expect(renderToString(tree)).toMatchSnapshot()
  })
})
