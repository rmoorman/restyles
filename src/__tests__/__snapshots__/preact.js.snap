// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restyles renders preact elements with styles 1`] = `"<div class=\\"css-vmubhh\\">This will be preact-purple until hovered.<div class=\\"profile\\">This font size will be 20px</div></div>"`;
